packages/xstate-graph/src/graph.ts

```diff
@@ -67,59 +73,49 @@ export function getChildren(stateNode: AnyStateNode): AnyStateNode[] {
   return children;
 }
 
-export function serializeState<TContext>(state: State<TContext, any>): string {
-  const { value, context } = state;
-  return context === undefined
-    ? JSON.stringify(value)
-    : JSON.stringify(value) + ' | ' + JSON.stringify(context);
+export function serializeMachineState(state: AnyState): SerializedState {
+  const { value, context, actions } = state;
+  return JSON.stringify({ value, context, actions }) as SerializedState;
 }
 
 export function serializeEvent<TEvent extends EventObject>(
   event: TEvent
-): string {
-  return JSON.stringify(event);
-}
-
-export function deserializeEventString<TEvent extends EventObject>(
-  eventString: string
-): TEvent {
-  return JSON.parse(eventString) as TEvent;
+): SerializedEvent {
+  return JSON.stringify(event) as SerializedEvent;
 }
 
 const defaultValueAdjMapOptions: Required<ValueAdjMapOptions<any, any>> = {
   events: {},
   filter: () => true,
-  stateSerializer: serializeState,
+  stateSerializer: serializeMachineState,
   eventSerializer: serializeEvent
 };
 
-function getValueAdjMapOptions<TContext, TEvent extends EventObject>(
-  options?: ValueAdjMapOptions<TContext, TEvent>
-): Required<ValueAdjMapOptions<TContext, TEvent>> {
+function getValueAdjMapOptions<TState, TEvent extends EventObject>(
+  options?: ValueAdjMapOptions<TState, TEvent>
+): Required<ValueAdjMapOptions<TState, TEvent>> {
   return {
     ...(defaultValueAdjMapOptions as Required<
-      ValueAdjMapOptions<TContext, TEvent>
+      ValueAdjMapOptions<TState, TEvent>
     >),
     ...options
   };
 }
 
-export function getAdjacencyMap<
-  TContext = DefaultContext,
-  TEvent extends EventObject = AnyEventObject
->(
-  node:
-    | StateNode<TContext, any, TEvent, any, any, any>
-    | StateMachine<TContext, any, TEvent, any, any, any, any>,
-  options?: ValueAdjMapOptions<TContext, TEvent>
-): AdjacencyMap<TContext, TEvent> {
+export function getAdjacencyMap<TMachine extends AnyStateMachine>(
```

Comment:
could we add a jsdoc comment above this that would explain what it does and what is it useful for? I only see some tests related to this within `@xstate/graph`'s test suite but the tests are mostly concerned about the shape of the returned object and I'm not exactly sure how to utilize that returned object creatively 

it would be cool to understand the difference between this util and other utils available in this pkg (some of which are actually used by `@xstate/test`)

Reply:
Yes, I will add the JS docs 👍

Reply:
Added as task above.